ros2topic/ros2topic/verb/echo.py

```diff
@@ -82,6 +84,59 @@ def add_arguments(self, parser, cli_name):
         parser.add_argument(
             '--raw', action='store_true', help='Echo the raw binary representation')
 
+    def chooseQoS(self, node, args):
+
+        qos_profile = qos_profile_from_short_keys(
+            args.qos_profile,
+            reliability=args.qos_reliability,
+            durability=args.qos_durability,
+            depth=args.qos_depth,
+            history=args.qos_history)
+
+        reliability_reliable_endpoints_count = 0
+        durability_transient_local_endpoints_count = 0
+        node_count = 0
+
+        try:
+            for info in node.get_publishers_info_by_topic(args.topic_name):
+                node_count += 1
+                if (info.qos_profile.reliability ==
+                    QoSReliabilityPolicy.RELIABLE):
+                    reliability_reliable_endpoints_count += 1
+                if (info.qos_profile.durability ==
+                    QoSDurabilityPolicy.TRANSIENT_LOCAL):
+                    durability_transient_local_endpoints_count += 1
+        except NotImplementedError as e:
+            return str(e)
+
+        # If all endpoints are reliable, ask for reliable
+        if reliability_reliable_endpoints_count == node_count:
+            qos_profile.reliability = QoSReliabilityPolicy.RELIABLE
+        else:
+            if reliability_reliable_endpoints_count > 0 :
+                print(
+                    'Some, but not all, publisher are offering '
+                    'QoSReliabilityPolicy.RELIABLE. Falling back to '
+                    'QoSReliabilityPolicy.BEST_EFFORT as it will connect '
+                    'to all publishers', end='\n\n'
+                )
+            qos_profile.reliability = QoSReliabilityPolicy.BEST_EFFORT
+
+        # If all endpoints are transient_local, ask for transient_local
+        if  durability_transient_local_endpoints_count == node_count:
```

Comment:
```suggestion
        if durability_transient_local_endpoints_count == node_count:
```

Reply:
Addresed in [8efa6ed](https://github.com/ros2/ros2cli/pull/613/commits/8efa6ede28b8f5a77c912a5ae690f0cf5db7ba75)